shared/cmd/flags.go

```diff
@@ -60,4 +60,15 @@ var (
 		Usage: "Indicate what fraction of p2p messages are sampled for tracing.",
 		Value: 0.20,
 	}
+	// DisableMonitoringFlag defines a flag to disable the metrics collection.
+	DisableMonitoringFlag = cli.BoolFlag{
+		Name:  "disable-monitoring",
+		Usage: "Disable monitoring service.",
+	}
+	// MonitoringEndpointFlag defines the http address and port used to serve prometheus metrics.
+	MonitoringEndpointFlag = cli.StringFlag{
+		Name:  "monitoring-endpoint",
```

Comment:
This is confusing. Can we just provide the monitoring port? cli.IntFlag , MetricsPort 

We expect the binary to serve `/metrics` on whatever port we provide here.
With the given default port of 8080, we can expect Prometheus to serve metrics at `http://0.0.0.0:8080/metrics`

Reply:
The http server expects a complete address to listen if you pass `:8080` the address is `0.0.0.0:8080`. If I put a flag for only the port I can't specify the address bind.
Do you think this is not needed?
https://golang.org/pkg/net/#Dial

Reply:
Right, but you don't have to require the user to do that. You can take the value of the flag and construct the string. `listenAddress := fmt.Sprintf(":%d", *metricsPortFlag)`

Reply:
I don't think the bind address is needed. We can reevaluate if that changes though